<Type Name="ErrorEventArgs" FullName="System.IO.ErrorEventArgs">
  <Metadata><Meta Name="ms.openlocfilehash" Value="b1d1bb3bca66aeda935cdbbe3fbc1826d95c7ab7" /><Meta Name="ms.sourcegitcommit" Value="8ae9a18b248625c8e1851e607c00346a64849587" /><Meta Name="ms.translationtype" Value="HT" /><Meta Name="ms.contentlocale" Value="ru-RU" /><Meta Name="ms.lasthandoff" Value="12/17/2019" /><Meta Name="ms.locfileid" Value="75132404" /></Metadata><TypeSignature Language="C#" Value="public class ErrorEventArgs : EventArgs" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi beforefieldinit ErrorEventArgs extends System.EventArgs" />
  <TypeSignature Language="DocId" Value="T:System.IO.ErrorEventArgs" />
  <TypeSignature Language="VB.NET" Value="Public Class ErrorEventArgs&#xA;Inherits EventArgs" />
  <TypeSignature Language="C++ CLI" Value="public ref class ErrorEventArgs : EventArgs" />
  <TypeSignature Language="F#" Value="type ErrorEventArgs = class&#xA;    inherit EventArgs" />
  <AssemblyInfo>
    <AssemblyName>System.IO.FileSystem.Watcher</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.1.1.0</AssemblyVersion>
    <AssemblyVersion>4.1.2.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.1.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.EventArgs</BaseTypeName>
  </Base>
  <Interfaces />
  <Docs>
    <summary><span data-ttu-id="d3fe9-101">Предоставляет данные для события <see cref="E:System.IO.FileSystemWatcher.Error" />.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-101">Provides data for the <see cref="E:System.IO.FileSystemWatcher.Error" /> event.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="d3fe9-102"><xref:System.IO.ErrorEventArgs> содержит <xref:System.Exception>, вызвавший событие <xref:System.IO.FileSystemWatcher.Error?displayProperty=nameWithType>.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-102"><xref:System.IO.ErrorEventArgs> contains the <xref:System.Exception> that caused the <xref:System.IO.FileSystemWatcher.Error?displayProperty=nameWithType> event.</span></span> <span data-ttu-id="d3fe9-103">Этот класс предоставляет <xref:System.IO.ErrorEventArgs.GetException%2A> метод для получения исключения.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-103">This class provides the <xref:System.IO.ErrorEventArgs.GetException%2A> method to retrieve the exception.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="d3fe9-104">В следующем примере создается новый экземпляр <xref:System.IO.ErrorEventArgs> и инициализируется с помощью <xref:System.Exception>.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-104">The following example creates a new instance of <xref:System.IO.ErrorEventArgs> and initializes it with an <xref:System.Exception>.</span></span> <span data-ttu-id="d3fe9-105">Затем в примере вызывается <xref:System.IO.ErrorEventArgs.GetException%2A> для получения <xref:System.Exception> и вывода сообщения об ошибке.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-105">Then the example calls <xref:System.IO.ErrorEventArgs.GetException%2A> to retrieve the <xref:System.Exception> and display the error message.</span></span> <span data-ttu-id="d3fe9-106">С этим кодом не связана ни одна форма.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-106">There is no form associated with this code.</span></span>  
  
 [!code-cpp[Classic ErrorEventArgs Example#1](~/samples/snippets/cpp/VS_Snippets_CLR_Classic/classic ErrorEventArgs Example/CPP/source.cpp#1)]
 [!code-csharp[Classic ErrorEventArgs Example#1](~/samples/snippets/csharp/VS_Snippets_CLR_Classic/classic ErrorEventArgs Example/CS/source.cs#1)]
 [!code-vb[Classic ErrorEventArgs Example#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_Classic/classic ErrorEventArgs Example/VB/source.vb#1)]  
  
 <span data-ttu-id="d3fe9-107">В следующем примере показано, как создать FileSystemWatcher для отслеживания изменений в файле (создание, удаление, переименование, изменение), происходящих на диске.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-107">The following example shows how to create a FileSystemWatcher to monitor file changes (creates, deletes, renames, changes) occurring on a disk drive.</span></span> <span data-ttu-id="d3fe9-108">В примере также показано, как правильно получать уведомления об ошибках.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-108">The example also shows how to properly receive error notifications.</span></span>  
  
 [!code-csharp[FileSystemWatcher#1](~/samples/snippets/csharp/VS_Snippets_CLR/FileSystemWatcher/cs/filesystemwatcher.cs#1)]
 [!code-vb[FileSystemWatcher#1](~/samples/snippets/visualbasic/VS_Snippets_CLR/FileSystemWatcher/VB/FileSystemWatcher.vb#1)]  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.IO.ErrorEventHandler" />
    <altmember cref="T:System.IO.FileSystemWatcher" />
    <altmember cref="M:System.IO.FileSystemWatcher.OnError(System.IO.ErrorEventArgs)" />
  </Docs>
  <Members>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ErrorEventArgs (Exception exception);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(class System.Exception exception) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.IO.ErrorEventArgs.#ctor(System.Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ErrorEventArgs(Exception ^ exception);" />
      <MemberSignature Language="F#" Value="new System.IO.ErrorEventArgs : Exception -&gt; System.IO.ErrorEventArgs" Usage="new System.IO.ErrorEventArgs exception" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.IO.FileSystem.Watcher</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.1.2.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="exception" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="exception"><span data-ttu-id="d3fe9-109">Компонент <see cref="T:System.Exception" />, представляющий возникшую ошибку.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-109">An <see cref="T:System.Exception" /> that represents the error that occurred.</span></span></param>
        <summary><span data-ttu-id="d3fe9-110">Инициализация нового экземпляра класса <see cref="T:System.IO.ErrorEventArgs" />.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-110">Initializes a new instance of the <see cref="T:System.IO.ErrorEventArgs" /> class.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Examples  
 <span data-ttu-id="d3fe9-111">В следующем примере создается новый экземпляр <xref:System.IO.ErrorEventArgs> и инициализируется с помощью <xref:System.Exception>.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-111">The following example creates a new instance of <xref:System.IO.ErrorEventArgs> and initializes it with an <xref:System.Exception>.</span></span> <span data-ttu-id="d3fe9-112">Затем в примере вызывается <xref:System.IO.ErrorEventArgs.GetException%2A> для получения <xref:System.Exception> и вывода сообщения об ошибке.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-112">Then the example calls <xref:System.IO.ErrorEventArgs.GetException%2A> to retrieve the <xref:System.Exception> and display the error message.</span></span> <span data-ttu-id="d3fe9-113">С этим кодом не связана ни одна форма.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-113">There is no form associated with this code.</span></span>  
  
 [!code-cpp[Classic ErrorEventArgs.ErrorEventArgs Example#1](~/samples/snippets/cpp/VS_Snippets_CLR_Classic/classic ErrorEventArgs.ErrorEventArgs Example/CPP/source.cpp#1)]
 [!code-csharp[Classic ErrorEventArgs.ErrorEventArgs Example#1](~/samples/snippets/csharp/VS_Snippets_CLR_Classic/classic ErrorEventArgs.ErrorEventArgs Example/CS/source.cs#1)]
 [!code-vb[Classic ErrorEventArgs.ErrorEventArgs Example#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_Classic/classic ErrorEventArgs.ErrorEventArgs Example/VB/source.vb#1)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="GetException">
      <MemberSignature Language="C#" Value="public virtual Exception GetException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig newslot virtual instance class System.Exception GetException() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.IO.ErrorEventArgs.GetException" />
      <MemberSignature Language="VB.NET" Value="Public Overridable Function GetException () As Exception" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; virtual Exception ^ GetException();" />
      <MemberSignature Language="F#" Value="abstract member GetException : unit -&gt; Exception&#xA;override this.GetException : unit -&gt; Exception" Usage="errorEventArgs.GetException " />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.IO.FileSystem.Watcher</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.1.2.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Exception</ReturnType>
      </ReturnValue>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="d3fe9-114">Получает компонент <see cref="T:System.Exception" />, представляющий возникшую ошибку.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-114">Gets the <see cref="T:System.Exception" /> that represents the error that occurred.</span></span></summary>
        <returns><span data-ttu-id="d3fe9-115">Компонент <see cref="T:System.Exception" />, представляющий возникшую ошибку.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-115">An <see cref="T:System.Exception" /> that represents the error that occurred.</span></span></returns>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Examples  
 <span data-ttu-id="d3fe9-116">В следующем примере создается новый экземпляр <xref:System.IO.ErrorEventArgs> и инициализируется с помощью <xref:System.Exception>.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-116">The following example creates a new instance of <xref:System.IO.ErrorEventArgs> and initializes it with an <xref:System.Exception>.</span></span> <span data-ttu-id="d3fe9-117">Затем в примере вызывается <xref:System.IO.ErrorEventArgs.GetException%2A> для получения <xref:System.Exception> и вывода сообщения об ошибке.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-117">Then the example calls <xref:System.IO.ErrorEventArgs.GetException%2A> to retrieve the <xref:System.Exception> and display the error message.</span></span> <span data-ttu-id="d3fe9-118">С этим кодом не связана ни одна форма.</span><span class="sxs-lookup"><span data-stu-id="d3fe9-118">There is no form associated with this code.</span></span>  
  
 [!code-cpp[Classic ErrorEventArgs.ErrorEventArgs Example#1](~/samples/snippets/cpp/VS_Snippets_CLR_Classic/classic ErrorEventArgs.ErrorEventArgs Example/CPP/source.cpp#1)]
 [!code-csharp[Classic ErrorEventArgs.ErrorEventArgs Example#1](~/samples/snippets/csharp/VS_Snippets_CLR_Classic/classic ErrorEventArgs.ErrorEventArgs Example/CS/source.cs#1)]
 [!code-vb[Classic ErrorEventArgs.ErrorEventArgs Example#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_Classic/classic ErrorEventArgs.ErrorEventArgs Example/VB/source.vb#1)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>
